remote_file_system/server.py

```diff
@@ -31,6 +32,8 @@ def __init__(self, server_ip_address: IPv4Address, server_port_number: int, file
         self.server_ip_address: IPv4Address = server_ip_address
         self.server_port_number: int = server_port_number
         self.server_file_system: ServerFileSystem = file_system
+        # store request id as key. value is the Message
+        self.message_history: Dict[uuid4, Message] = defaultdict(list)
```

Comment:
I think this should be initialised to an empty dictionary with `{}`, instead of `defaultdict(list)` since we are storing a single message not a list of messages associated with a UUID.
The type annotation should be the `UUID` class, since `uuid4` is a method and not a valid type.
```suggestion
        # store request id as key. value is the Message
        self.message_history: Dict[UUID, Message] = {}
```

Reply:
sure